pkg/cloud/vsphere/services/cloudprovider/csi.go

```diff
@@ -138,7 +148,7 @@ func CSIDriver() *storagev1beta1.CSIDriver {
 	}
 }
 
-func VSphereCSINodeDaemonSet(storageConfig cloudprovider.StorageConfig) *appsv1.DaemonSet {
+func VSphereCSINodeDaemonSet(storageConfig *cloudprovider.StorageConfig) *appsv1.DaemonSet {
```

Comment:
Why pass by reference? Unless changes are made, passing by copy is actually a lot easier on Go.

Reply:
It's cause `StorageConfig` is now a pointer, otherwise we can't do nil checks to disable the storage config

Reply:
You can always dereference when passing in.

Reply:
good point